src/storage/kura/model.rs

```diff
@@ -0,0 +1,216 @@
+/// This module contains core `Kura` stuctures.
+
+/// Transaction data is permanently recorded in files called blocks. Blocks are organized into
+/// a linear sequence over time (also known as the block chain).
+//TODO: based on https://iroha.readthedocs.io/en/latest/concepts_architecture/glossary.html#block
+//signatures placed outside of the payload - should we store them?
+#[derive(Clone, serde::Serialize, serde::Deserialize)]
+pub struct Block {
+    /// a number of blocks in the chain up to the block.
+    pub height: u64,
+    /// Unix time (in milliseconds) of block forming by a peer.
+    pub timestamp: u64,
+    /// array of transactions, which successfully passed validation and consensus step.
+    pub transactions: Vec<Transaction>,
+    /// hash of a previous block in the chain.
+    pub previous_block_hash: Hash,
+    /// rejected transactions hashes — array of transaction hashes, which did not pass stateful
+    /// validation step; this field is optional.
+    pub rejected_transactions_hashes: Option<Vec<Hash>>,
+}
+
+impl std::convert::From<Block> for Vec<u8> {
+    fn from(block: Block) -> Self {
+        bincode::serialize(&block).expect("Failed to serialize block.")
+    }
+}
+
+#[derive(Clone, Debug, serde::Serialize, serde::Deserialize)]
+pub struct Hash {}
+
+pub struct Account {
+    /// identifier of an account. Formatted as `account_name@domain_id`.
+    id: String,
+}
+
+pub struct AccountHasAsset {
+    account_id: String,
+    asset_id: String,
+    amount: u64,
+}
+
+#[derive(Clone)]
+pub struct Asset {
+    /// identifier of asset, formatted as asset_name#domain_id
+    pub id: String,
+}
+
+/// An ordered set of commands, which is applied to the ledger atomically.
+#[derive(Clone, serde::Serialize, serde::Deserialize)]
+pub struct Transaction {
+    /// An ordered set of commands.
+    pub commands: Vec<Command>,
+    /// Time of creation (unix time, in milliseconds).
+    pub creation_time: u64,
+    /// Account ID of transaction creator (username@domain).
+    pub account_id: String,
+    /// Quorum field (indicates required number of signatures).
+    pub quorum: u8,
+    pub signatures: Vec<Signature>,
+}
+
+#[derive(Clone, serde::Serialize, serde::Deserialize)]
+pub struct Signature {}
+
+/// A command is an intention to change the state of the network.
+/// For example, in order to create a new role in Iroha you have to issue Create role command.
+#[derive(Clone, serde::Serialize, serde::Deserialize)]
+pub struct Command {
+    version: u8,
+    command_type: u8,
```

Comment:
What is the mapping between `command_type` and actual commands like `AddAssetQuantity`? Maybe it's better to use **enum** here?

Reply:
@eadventurous - good point.
Unfortunately for us and good for clients they can create custom commands and we will need to support them.

Reply:
Yes, I remember that requirement, in that case maybe something like:
```
pub enum Command {
    AddAssetQuantity(commands::AddAssetQuantity),
    ...
    Custom(Vec<u8>)
}
```

Reply:
Let's think about it. We need to find cases where enum will be helpful when we work with commands.